elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLTimeDimension.java

```diff
@@ -40,5 +44,17 @@ public SQLTimeDimension(Table table, String fieldName, EntityDictionary dictiona
             this.reference = generateColumnReference(path, dictionary);
             this.joinPath = path;
         }
+
+        this.metadataDictionary = dictionary;
+    }
+
+    @Override
+    public Pair<String, String> getSourceTableAndColumn() {
```

Comment:
I'm not a huge fan of Pair as a function result.  Another option - not sure which one I like better yet.  We could override getSourceColumn.  

Rather than a two pass building phase, getSourceColumn can lazy initialize:
```java
    Column getSourceColumn() {
        if (joinColumn == null || sourceColumn != null) {
            return sourceColumn;
        }

        sourceColumn = getJoinPath()....  //Do lookup logic here
        return sourceColumn;
    }
```

Reply:
Currently the `JoinTo` concept is only in SQLQueryEngine, so that `joinColumn` doesn't exist in the common aggregation data store. I think we don't have to avoid two pass building phase now.